vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -139,6 +139,12 @@ def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
                 )
             else:
                 raise NotImplementedError("CPU MOE only supports x86 arch.")
+        if current_platform.is_hpu():
+            for expert_id in range(layer.local_num_experts):
+                layer.moe_op.w13_list[expert_id].set_weight(
+                    layer.w13_weight.data[expert_id])
+                layer.moe_op.w2_list[expert_id].set_weight(
+                    layer.w2_weight.data[expert_id])
```

Comment:
Moved weight setting from _load_w13 and _load_w2 to process_weights_after_loading.
This works better for both llama4, qwen3 and deepseek